services/galley/src/Galley/API/LegalHold.hs

```diff
@@ -356,4 +362,45 @@ disableForUser zusr tid uid (Public.DisableLegalHoldForUserRequest mPassword) =
       -- TODO: send event at this point (see also: related TODO in this module in
       -- 'approveDevice' and
       -- https://github.com/wireapp/wire-server/pull/802#pullrequestreview-262280386)
-      LegalHoldData.setUserLegalHoldStatus tid uid UserLegalHoldDisabled
+      changeLegalholdStatus tid uid UserLegalHoldDisabled
+
+changeLegalholdStatus :: TeamId -> UserId -> UserLegalHoldStatus -> Galley ()
+changeLegalholdStatus tid uid lhStatus = do
```

Comment:
i think this is almost the right approach, but instead we should have a function `connectionLHConflict :: UserId -> UserId -> m Bool`, and later one `groupLHConflict :: UserId -> ConvId -> m Bool`.

Or maybe we can also just have one function, and the ConvId either points to a 1:1 or a group, but this function doesn't care.

Anyway, we'll need to call this function when handling a LH device request, before setting the device status to pending.  But we'll also need this when handling user connect requests, before setting the connection to pending.  I think we can use the same helper for both (at least that would be nice).

Reply:
not sure the above is making sense, but here is another thought: we may not be able to run this synchronously, we may need to move blocking to a background queue.  i think the resulting race conditions are ok: security is also provided on a per-device level later, and it's unlikely to happen, since the user under LH still has to approve the request before the device is activated.